src/lualib/Error.ts

```diff
@@ -0,0 +1,93 @@
+Error = class {
+    public static captureStackTrace(targetObject: {}, depth: number): void {
+        targetObject["stack"] = debug.traceback(undefined, depth);
+    }
+
+    public message: string;
+    public name: string;
+    public stack?: string;
+
+    constructor(message?: string, name?: string) {
+        this.message = message || "";
+        this.name = name || "Error";
+        Error["captureStackTrace"](this, name ? 5 : 4);
+    }
+
+    public __tostring(): string {
+        return `${this.name}: ${this.message}`;
+    }
+} as any;
+
+setmetatable(Error, { __call: (_self: any, message: string) => new Error(message) });
+
+/** Standard error types */
+
+RangeError = class extends Error {
+    constructor(message?: string) {
+        // @ts-ignore
+        super(message, "RangeError");
+    }
+    public __tostring(): string {
```

Comment:
Would custom subclasses also have to redefine it?

Reply:
Unfortunately yes, as Lua basically uses `rawget` to find metamethods.

Reply:
Hmm, maybe `getmetatable(this).__tostring = ...` in Error constructor?

Reply:
Hm that would work for now.